docs/latest/design/envoy-patch-policy.md

```diff
@@ -0,0 +1,150 @@
+# EnvoyPatchPolicy
+
+## Overview
+
+This design introduces the `EnvoyPatchPolicy` API allowing users to modify the generated Envoy xDS Configuration
+that Envoy Gateway generates before sending it to Envoy Proxy.
+
+Envoy Gateway allows users to configure networking and security intent using the
+upstream [Gateway API][] as well as implementation specific [Extension APIs][] defined in this project
+to provide a more batteries included experience for application developers.
+* These APIs are an abstracted version of the underlying Envoy xDS API to provide a a better user experience for the application developer, exposing and setting only a subset of the fields for a specific feature, sometimes in a opinionated way (e.g [RateLimit][])
+* These APIs do not expose all the features capabilities that Envoy has either because these features are desired but the API
+is not defined yet or the project cannot support such an extensive list of features.
+To alleviate this problem, and provide an interim solution for a small section of advanced users who are well versed in
+Envoy xDS API and its capabilities, this API is being introduced.
+
+## Goals
+* Add an API allowing users to modify the generated xDS Configuration
+
+## Non Goals
+* Support multiple patch mechanisims
+
+## Implementation
+`EnvoyPatchPolicy` is a [Direct Policy Attachment][] type API that can be used to extend [Gateway API][]
+Modifications to the generated xDS configuration can be provided as a JSON Patch which is defined in
+[RFC 6902][]. This patching mechanism has been adopted in [Kubernetes][] as well as [Kustomize][] to update
+resource objects.
+
+### Example
+Here is an example highlighting how a user can configure global ratelimiting using an external rate limit service using this API.
```

Comment:
Given that we can already configure rate limiting using the native Envoy Gateway ratelimit API, I'd like to see some other examples:

1. You have three different upstream services, each of which needs to use a specific certificate to originate TLS.
   - The current Gateway API doesn't have a lot of support for upstream TLS origination. There's ongoing work to support it natively, but until that's accepted, this would be a reasonable use for an extension mechanism.
   - Obviously you don't know the routes involved, you just know that you happen to have three upstream services that need certificates.

1. You need to allow the HTTP listener, but not the HTTPS listener, to route traffic to the path prefix `/.well-known/acme-challenge/` over to the `my-acme-server` Service. (Other traffic may arrive on the HTTP listener too, but that traffic should be unaffected.)
   - This is the path used by the HTTP-01 ACME challenge, which always arrives over HTTP and must not be redirected to HTTPS. If a vendor wants to do custom ACME work, this might be a thing they'd need to do.

Reply:
can you share some raw xds Config for the examples ?